kuksa-client/kuksa_client/grpc/aio.py

```diff
@@ -262,8 +267,15 @@ async def set(self, *, updates: Collection[EntryUpdate], **rpc_kwargs) -> None:
             raise VSSClientError.from_grpc_error(exc) from exc
         self._process_set_response(resp)
 
-    async def authorize(self, *, token: str) -> str:
-        raise NotImplementedError('"authorize" is not yet implemented')
+    async def authorize(self, *, token: Dict) -> None:
+        self.authorization_header = "Bearer " + token
+        if self.rpc_metadata != None:
+            metadata = dict(self.rpc_metadata)
+        else: 
+            metadata = dict()
+        metadata = metadata | {"authorization": self.authorization_header}
+        self.rpc_metadata = list(metadata.items())
+
```

Comment:
I'm not sure this change is resolving anything. The function most probably should continue accepting a token of type `str` and not a `Dict`.

I'm also not sure why `self.rpc_metadata` is needed, or that it even makes sense. See my other comment below.

Reply:
I agree, why I used this is. If we provide the string from the other function the string looks like ```{'token', 'ajsjdksjdaksdj'}``` so in order to get the token out I used a ```Dict```. Else it would be needed to split the string or something like that or fix it in the fucntion that gets the token from the file. I didn't change the function for getting token, because I thought Romaine had some genius thoughts behind it :)

Reply:
I intorduced ```self.rpc_metadata``` because I thought we do not need to check the metadata in every call just when setting a new token. Like you described I was wrong.

Reply:
> I intorduced self.rpc_metadata because I thought we do not need to check the metadata in every call just when setting a new token.

Ah, now I understand (I think).

So yes, if no `self.authorization_header` is set, no header should be added to the request. What I couldn't understand was all the extra code & variables added. A simple:

```
if self.authorization_header:
    // Add authorization header to request
```

would accomplish that.

Also change the `__init__` function to:

```
self.authorization_header = None
```